api.js

```diff
@@ -258,17 +259,42 @@ Api.prototype.mediaPut = function (req, res, m, q) {
     return
```

Comment:
We should check first that q.file and q.thumbnail are not undefined, and then we should check for both of their existence. The status code should be 400, since this is a bad request – 404 would indicate that the route is not found.

Reply:
Agreed re 400 vs 404. Done.